ci/telemetry_logging.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Telemetry Logging tests
+  hosts:  "{{ cifmw_target_hook_host | default('localhost')  }}"
+  gather_facts: true
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  vars_files:
+    - vars/common.yml
+    - vars/osp18_env.yml
+  tasks:
+    - name: "Run logging Tests"
+      ansible.builtin.import_role:
+        name: telemetry_logging
+
+        - import_role:
+            name: telemetry_logging
+            tasks_from: main
```

Comment:
The default value is "main"

I would suggest refactoring the main.yml to have two sections.
For all compute-node tests, use "when: telemetry_logging_run_compute_tests | bool"
For all controller tests, use "when: telemetry_logging_run_controller_tests | bool"

This means you can have separate plays in this playbook, which import the same role, but pass different vars.

e.g. there would be multiple plays in this playbook.

```
- name: run controller tests
  hosts: controller_nodes # please change to the right value
  tasks:
    - import_role:
          name: telemetry_logging
          vars:
                telemetry_logging_run_compute_tests: true

- name: Run compute tests
  hosts: compute_nodes ## check the name
  tasks:
      - import_role:
          name: telemetry_logging
          vars:
              telemetry_logging_run_controller_tests: true
```

The info in this suggestion is incomplete, you will need to test it, and add the appropriate missing information

Reply:
ack